lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -90,11 +37,23 @@ const useFetchGrade = (student, setFetchGradeError) => {
  * @param {SubmitGradeFormProps} props
  */
 export default function SubmitGradeForm({ student }) {
-  // State for loading the grade
-  const [fetchGradeError, setFetchGradeError] = useState(
-    /** @type {ErrorLike|null} */ (null)
+  const [fetchGradeErrorDismissed, setFetchGradeErrorDismissed] =
+    useState(false);
+  const gradingService = useService(GradingService);
+  const grade = useFetch(
+    student ? `grade:${student.userid}` : null,
+    student
+      ? async () => {
+          setFetchGradeErrorDismissed(false);
+          const { currentScore = null } = await gradingService.fetchGrade({
+            student,
+          });
+          return currentScore === null
+            ? ''
+            : `${scaleGrade(currentScore, GRADE_MULTIPLIER)}`;
+        }
+      : undefined
```

Comment:
The nested ternary expressions in a function invocation make this fairly cumbersome to scan, though I can see why they are convenient at the same time. Would it be possible to assign the `fetcher` function to a reference (here `gradeFetcher`) such that this could be:

```suggestion
  const grade = useFetch(
    student ? `grade:${student.userid}` : null,
    student
      ? gradeFetcher
      : undefined
```

?

Reply:
Makes sense. I'll do this.